src/Gateways/AbstractGateway.php

```diff
@@ -852,6 +898,21 @@ protected function handle_response( Requests\RequestInterface $request, Transact
 
 			return false;
 		}
+		
+		//reverse incase of AVS/CVN failure
+		if(!empty($response->transactionReference->transactionId) && !empty($this->check_avs_cvv)){
```

Comment:
I had to change this line as follows for the logic to work properly during testing
```suggestion
		if(!empty($response->transactionReference->transactionId) && $this->get_option('check_avs_cvv') === 'yes'){
```

Reply:
conditions updated